src/Module/Api/Mastodon/FollowRequests.php

```diff
@@ -30,32 +28,34 @@ public static function init(array $parameters = [])
 	 * @param array $parameters
 	 * @throws HTTPException\BadRequestException
 	 * @throws HTTPException\ForbiddenException
+	 * @throws HTTPException\InternalServerErrorException
 	 * @throws HTTPException\NotFoundException
 	 * @throws HTTPException\UnauthorizedException
+	 * @throws \ImagickException
+	 *
 	 * @see https://docs.joinmastodon.org/methods/accounts/follow_requests#accept-follow
 	 * @see https://docs.joinmastodon.org/methods/accounts/follow_requests#reject-follow
 	 */
 	public static function post(array $parameters = [])
 	{
 		parent::post($parameters);
 
-		$Intro = DI::intro()->fetch(['id' => $parameters['id'], 'uid' => self::$current_user_id]);
-
-		$contactId = $Intro->{'contact-id'};
+		$Introduction = DI::intro()->selectFirst(['id' => $parameters['id'], 'uid' => self::$current_user_id]);
 
-		$relationship = new Mastodon\Relationship();
-		$relationship->id = $contactId;
+		$contactId = $Introduction->{'contact-id'};
 
 		switch ($parameters['action']) {
 			case 'authorize':
-				$Intro->confirm();
-				$relationship = Mastodon\Relationship::createFromContact(Contact::getById($contactId));
+				DI::intro()->confirm($Introduction);
+				$relationship = new Relationship($contactId, Contact::getById($contactId));
```

Comment:
That's the main, main, main reason for the `Factory` pattern. Imagine I want to test this function and mock a custom relationship to test how the business logic is reacting on it.
=> This isn't possible now.

But if we would have
```suggestion
				$relationship = DI::factoryRelationship()->create($contactId, Contact::getById($contactId));
```
I would mock the `DI::factoryRelationship()` call and return custom relationships like I do want :-).

Reply:
Got it. I skipped the Relationship repository because it looked straightforward but I see what you mean.